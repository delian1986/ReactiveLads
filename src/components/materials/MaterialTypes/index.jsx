import { Component } from "react";
import { MaterialType } from "../MaterialType";
import PropTypes from "prop-types";

export class MaterialTypes extends Component {
  render() {
    const {
      materialTypes,
      selectMaterialType,
      selectedMaterialsFilter
    } = this.props;
    return (
      <>
        <h5>Material Types</h5>
        {materialTypes.length > 0 ? (
          <div>
            {materialTypes.map((materialType) => (
              <MaterialType
                key={materialType.id}
                id={materialType.id}
                name={materialType.name}
                selectMaterialType={selectMaterialType}
                isSelected={selectedMaterialsFilter.some(
                  (typeId) => typeId === materialType.id
                )}
              />
            ))}
          </div>
        ) : (
          <div className="d-flex justify-content-center">
            <div className="spinner-border" role="status">
              <span className="sr-only">Loading...</span>
            </div>
          </div>
        )}
      </>
    );
  }
}

MaterialTypes.propTypes = {
  materialTypes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      selectMaterialType: PropTypes.func
    })
  ),
  selectMaterialType: PropTypes.func.isRequired,
  selectedMaterialsFilter: PropTypes.array
};
