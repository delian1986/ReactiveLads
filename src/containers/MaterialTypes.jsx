import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { MaterialTypes } from "../components/materials/MaterialTypes/index.jsx";

import fetchMaterialTypesThunk from "../services/fetchMaterialTypesThunk";
import { selectMaterialType } from "../actions/filters";

const mapStateToProps = (state) => {
  return {
    materialTypes: state.materialTypes
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchMaterialTypes: fetchMaterialTypesThunk,
      selectMaterialType: selectMaterialType
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(MaterialTypes);
