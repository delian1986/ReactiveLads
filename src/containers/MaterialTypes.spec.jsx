import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import MaterialTypes from "./MaterialTypes";

const mockStore = configureStore([]);

describe("tests <MaterialTypes/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      materialTypes: [
        {
          id: 1,
          name: "data",
          fetchMaterialTypes: jest.fn(),
          selectMaterialType: jest.fn()
        }
      ],
      filters: {
        selectedMaterialTypes: [3, 4]
      }
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <MaterialTypes component={() => <></>} />
        </Provider>
      </BrowserRouter>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("Actions", () => {
    it("should dispatch an action on button click", () => {
      renderer.act(() => {
        component.root.findByType("input").props.onChange();
      });

      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
  });
});
