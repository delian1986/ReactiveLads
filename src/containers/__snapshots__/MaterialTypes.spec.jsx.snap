// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <MaterialTypes/> component Test rendering component should render with given state from Redux store 1`] = `
Array [
  <h5>
    Material Types
  </h5>,
  <div>
    <div>
      <label>
        <input
          checked={false}
          className="mr-2"
          onChange={[Function]}
          type="checkbox"
        />
        data
      </label>
    </div>
  </div>,
]
`;
